lib/interceptor.js

```diff
@@ -94,6 +107,7 @@ Interceptor.prototype.reply = function reply(statusCode, body, rawHeaders) {
 
   if (this.scope._defaultReplyHeaders) {
     headers = headers || {}
+    // because of this, this.rawHeaders gets lower-cased versions of the `rawHeaders` param. is that a bug?
```

Comment:
The `headers` object is meant to be case-insensitive by always holding the lowercase version of the keys, however, the case-sensitivity of `rawHeaders` is inconsistent. 
Usually they're kept as the provided value, but sometimes (like here) they get lowercased first. 
I'm not positive what the intent is, I've just added a couple comments about the inconsistency when I come across them.

Reply:
Ah I see. Makes sense: you're saying we want `rawHeaders` to keep their original case.

Reply:
I don't use `rawHeaders` in any of my projects and the README never mentions them. So I'm not sure what we want.
If the intention is to mimic [`http.IncomingMessage.rawHeaders`](https://nodejs.org/api/http.html#http_message_rawheaders) then it seems the current implementation is buggy.

> The raw request/response headers list exactly as they were received.
...
Header names are not lowercased, and duplicates are not merged. 
...
For example, the previous message header object might have a rawHeaders list like the following:
```js
[ 'ConTent-Length', '123456',
  'content-LENGTH', '123',
  'content-type', 'text/plain',
  'CONNECTION', 'keep-alive',
  'Host', 'mysite.com',
  'accepT', '*/*' ]
```
This topic should probably be moved to its own issue if we consider it a bug.

Reply:
> If the intention is to mimic [`http.IncomingMessage.rawHeaders`](https://nodejs.org/api/http.html#http_message_rawheaders) then it seems the current implementation is buggy.

Agreed, let's open a new issue. It sounds like it may require some thinking through.

We do want the `rawHeaders` on a nock-produced IncomingMessage to match what they would be on a real one, and ideally it should be possible to establish through nock whatever casing the caller might want.

Interestingly, I don't think that is the format of our `rawHeaders` object. I think at least one of our tests for rawHeaders is showing an array for the duplicated headers:

```
[
  'content-length': ['123456', '123']
]
```